store.go

```diff
@@ -701,6 +713,28 @@ func (s *store) load() error {
 	return nil
 }
 
+func (s *store) LockDigest(d digest.Digest) error {
+	lockfile, err := GetLockfile(filepath.Join(s.lockRoot, d.Encoded()))
+	if err != nil {
+		return err
+	}
+	lockfile.Lock()
+	return nil
+}
+
+func (s *store) UnlockDigest(d digest.Digest) error {
+	lockPath := filepath.Join(s.lockRoot, d.Encoded())
+	lockfile, err := GetLockfile(lockPath)
+	if err != nil {
+		return err
+	}
+	// this might be called multiple times
```

Comment:
?

The conventional wisdom is that trying to unlock a not-locked lock is indicative of incorrect state tracking in the caller, and risks to unlock a lock owned by _some other process_.

Looking at https://github.com/containers/image/pull/607, this is intentional (and I haven’t read that PR at all yet) — but eventually the reason for this design, and the conditions when it is/isn’t save to use the API that way, should be documented within this project as well.

Reply:
> The conventional wisdom is that trying to unlock a not-locked lock is indicative of incorrect state tracking in the caller, and risks to unlock a lock owned by some other process.

I think we can avoid that with some extra bookkeeping in the containers/image.

Thanks for the comment!

Reply:
I removed that check, so we'll properly fail.

Reply:
This was intended as a request for documentation at this point, but removing this behavior works too :)